cmd/register/files.go

```diff
@@ -0,0 +1,179 @@
+	}
+	logger.Debugf(ctx, "Failed to unmarshal file %v for launch plan type", fname)
+	return nil, errors.New(fmt.Sprintf("Failed unmarshalling file %v", fname))
+
+}
+
+func register(ctx context.Context, message proto.Message, name string, cmdCtx cmdCore.CommandContext) error {
+	switch message.(type) {
+	case *admin.LaunchPlanSpec:
+		_, err := cmdCtx.AdminClient().CreateLaunchPlan(ctx, &admin.LaunchPlanCreateRequest{
+			Id: &core.Identifier{
+				ResourceType: core.ResourceType_LAUNCH_PLAN,
+				Project:      config.GetConfig().Project,
+				Domain:       config.GetConfig().Domain,
+				Name:         name,
+				Version:      GetConfig().version,
+			},
+			Spec: message.(*admin.LaunchPlanSpec),
+		})
+		return err
+	case *admin.WorkflowSpec:
+		_, err := cmdCtx.AdminClient().CreateWorkflow(ctx, &admin.WorkflowCreateRequest{
+			Id: &core.Identifier{
+				ResourceType: core.ResourceType_WORKFLOW,
+				Project:      config.GetConfig().Project,
+				Domain:       config.GetConfig().Domain,
+				Name:         name,
+				Version:      GetConfig().version,
+			},
+			Spec: message.(*admin.WorkflowSpec),
+		})
+		return err
+	case *admin.TaskSpec:
+		_, err := cmdCtx.AdminClient().CreateTask(ctx, &admin.TaskCreateRequest{
+			Id: &core.Identifier{
+				ResourceType: core.ResourceType_TASK,
+				Project:      config.GetConfig().Project,
+				Domain:       config.GetConfig().Domain,
+				Name:         name,
+				Version:      GetConfig().version,
+			},
+			Spec: message.(*admin.TaskSpec),
+		})
+		return err
+	default:
+		return errors.New(fmt.Sprintf("Failed registering unknown entity  %v type as part of %v file", message, name))
+	}
+}
+
+func hydrateNode(node *core.Node) {
+	targetNode := node.Target
+	switch targetNode.(type) {
+	case *core.Node_TaskNode:
+		taskNodeWrapper := targetNode.(*core.Node_TaskNode)
+		taskNodeReference := taskNodeWrapper.TaskNode.Reference.(*core.TaskNode_ReferenceId)
+		hydrateIdentifier(taskNodeReference.ReferenceId)
+	case *core.Node_WorkflowNode:
+		workflowNodeWrapper := targetNode.(*core.Node_WorkflowNode)
+		switch workflowNodeWrapper.WorkflowNode.Reference.(type) {
+		case *core.WorkflowNode_SubWorkflowRef:
+			subWorkflowNodeReference := workflowNodeWrapper.WorkflowNode.Reference.(*core.WorkflowNode_SubWorkflowRef)
+			hydrateIdentifier(subWorkflowNodeReference.SubWorkflowRef)
+		case *core.WorkflowNode_LaunchplanRef:
+			launchPlanNodeReference := workflowNodeWrapper.WorkflowNode.Reference.(*core.WorkflowNode_LaunchplanRef)
+			hydrateIdentifier(launchPlanNodeReference.LaunchplanRef)
+		}
+	}
+}
+
+func hydrateIdentifier(identifier *core.Identifier) {
+	identifier.Project = config.GetConfig().Project
+	identifier.Domain = config.GetConfig().Domain
+	identifier.Version = GetConfig().version
+}
+
+func hydrateSpec(message proto.Message) {
+	switch message.(type) {
+	case *admin.LaunchPlanSpec:
+		launchSpec := message.(*admin.LaunchPlanSpec)
+		hydrateIdentifier(launchSpec.WorkflowId)
```

Comment:
this should now be LaunchPlan

Reply:
Done . And tested this with flytekit 0.16.0b2